packages/serverless-contracts/src/contracts/apiGateway/features/lambdaHandler.ts

```diff
@@ -52,63 +52,64 @@ export const getApiGatewayHandler =
     AuthorizerType,
     Output,
     AdditionalArgs
-  > =>
-  async (event, context, callback, ...additionalArgs) => {
-    try {
-      const ajv = new Ajv();
+  > => {
+    const ajv = new Ajv();
+    const inputValidator = ajv.compile(contract.inputSchema);
 
-      const parsedEvent = proxyEventToHandlerEvent<
-        IntegrationType,
-        AuthorizerType,
-        PathParameters,
-        QueryStringParameters,
-        Headers,
-        CustomRequestContext,
-        Body
-      >(event);
+    return async (event, context, callback, ...additionalArgs) => {
+      try {
+        const parsedEvent = proxyEventToHandlerEvent<
+          IntegrationType,
+          AuthorizerType,
+          PathParameters,
+          QueryStringParameters,
+          Headers,
+          CustomRequestContext,
+          Body
+        >(event);
 
-      const inputValidator = ajv.compile(contract.inputSchema);
-      if (!inputValidator(parsedEvent)) {
-        console.error('Error: Invalid input');
-        console.error(inputValidator.errors);
-        throw createHttpError(400, 'Invalid input');
-      }
+        if (!inputValidator(parsedEvent)) {
+          console.error('Error: Invalid input');
+          console.error(inputValidator.errors);
+          throw createHttpError(400, 'Invalid input');
+        }
 
-      const handlerResponse = await handler(
-        parsedEvent,
-        context,
-        callback,
-        ...additionalArgs,
-      );
+        const handlerResponse = await handler(
+          parsedEvent,
+          context,
+          callback,
+          ...additionalArgs,
+        );
 
-      if (contract.outputSchema !== undefined) {
-        const outputValidator = ajv.compile(contract.outputSchema);
-        if (!outputValidator(handlerResponse)) {
-          console.error('Error: Invalid output');
-          console.error(outputValidator.errors);
-          throw createHttpError(400, 'Invalid output');
+        if (contract.outputSchema !== undefined) {
+          const outputValidator = ajv.compile(contract.outputSchema);
```

Comment:
You can move this out of the handler declaration as well right?

Reply:
Nicely spotted 👀